granulate_utils/metadata/cloud.py

```diff
@@ -151,6 +161,23 @@ def get_azure_metadata() -> Optional[AzureInstanceMetadata]:
     )
 
 
+def get_oracle_cloud_metadata() -> Optional[OracleCloudInstanceMetadata]:
+    # Documentation: https://docs.oracle.com/en-us/iaas/Content/Compute/Tasks/gettingmetadata.htm
+    response = send_request("http://169.254.169.254/opc/v2/instance/", headers={"Authorization": "Bearer Oracle"})
+    if response is None:
+        return None
+    instance = response.json()
+
+    return OracleCloudInstanceMetadata(
+        provider="oracle",
+        instance_type=instance["shape"],
+        region=instance["canonicalRegionName"],
+        memory=instance["shapeConfig"]["memoryInGBs"],
+        name=instance["hostname"],
```

Comment:
For GCP we access `name` and not `hostname`. Find out if it matters and make the behavior here consistent.